keras_nlp/models/bert/bert_feature_extractor.py

```diff
@@ -0,0 +1,269 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""BERT feature extraction model."""
+
+import copy
+import os
+
+from tensorflow import keras
+
+from keras_nlp.models.bert.bert_backbone import BertBackbone
+from keras_nlp.models.bert.bert_backbone import bert_kernel_initializer
+from keras_nlp.models.bert.bert_preprocessor import BertPreprocessor
+from keras_nlp.models.bert.bert_presets import backbone_presets
+from keras_nlp.utils.pipeline_model import PipelineModel
+from keras_nlp.utils.python_utils import classproperty
+from keras_nlp.utils.python_utils import format_docstring
+
+PRESET_NAMES = ", ".join(list(backbone_presets))
+
+
+@keras.utils.register_keras_serializable(package="keras_nlp")
+class BertFeatureExtractor(PipelineModel):
+    """An end-to-end BERT model for feature extraction
+
+    For usage of this model with pre-trained weights, see
```

Comment:
Given the understandable confusion below we should probably leave a comment about how this can only be used as a submodel and not directly compiled and fit. We could include a classification example on top of it for clarity.